pkg/sdk/testint/functions_integration_test.go

```diff
@@ -0,0 +1,407 @@
+		require.NoError(t, err)
+	})
+
+	t.Run("alter function: set secure", func(t *testing.T) {
+		f := createFunctionForSQLHandle(t, true)
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, f.Name)
+		set := sdk.NewFunctionSetRequest().WithSecure(sdk.Bool(true))
+		request := sdk.NewAlterFunctionRequest(id).WithArgumentTypes(defaultArgumentTypes).WithSet(set)
+		err := client.Functions.Alter(ctx, request)
+		require.NoError(t, err)
+
+		e, err := client.Functions.ShowByID(ctx, id)
+		require.NoError(t, err)
+		require.Equal(t, true, e.IsSecure)
+	})
+
+	t.Run("alter function: unset secure", func(t *testing.T) {
+		f := createFunctionForSQLHandle(t, true)
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, f.Name)
+		unset := sdk.NewFunctionUnsetRequest().WithSecure(sdk.Bool(true))
+		request := sdk.NewAlterFunctionRequest(id).WithArgumentTypes(defaultArgumentTypes).WithUnset(unset)
+		err := client.Functions.Alter(ctx, request)
+		require.NoError(t, err)
+
+		e, err := client.Functions.ShowByID(ctx, id)
+		require.NoError(t, err)
+		require.Equal(t, false, e.IsSecure)
+	})
+
+	t.Run("alter function: set and unset tags", func(t *testing.T) {
+		f := createFunctionForSQLHandle(t, true)
+
+		id := sdk.NewSchemaObjectIdentifier(databaseTest.Name, schemaTest.Name, f.Name)
+		setTags := []sdk.TagAssociation{
+			{
+				Name:  tagTest.ID(),
+				Value: "abc",
+			},
+		}
+		request := sdk.NewAlterFunctionRequest(id).WithArgumentTypes(defaultArgumentTypes).WithSetTags(setTags)
+		err := client.Functions.Alter(ctx, request)
+		require.NoError(t, err)
+
+		unsetTags := []sdk.ObjectIdentifier{
+			tagTest.ID(),
+		}
+		request = sdk.NewAlterFunctionRequest(id).WithArgumentTypes(defaultArgumentTypes).WithUnsetTags(unsetTags)
+		err = client.Functions.Alter(ctx, request)
+		require.NoError(t, err)
+	})
+
+	t.Run("show function for SQL: without like", func(t *testing.T) {
+		f1 := createFunctionForSQLHandle(t, true)
+		f2 := createFunctionForSQLHandle(t, true)
+
+		functions, err := client.Functions.Show(ctx, sdk.NewShowFunctionRequest())
+		require.NoError(t, err)
+
+		require.Equal(t, 2, len(functions))
+		require.Contains(t, functions, *f1)
+		require.Contains(t, functions, *f2)
+	})
+
+	t.Run("show function for SQL: with like", func(t *testing.T) {
+		f1 := createFunctionForSQLHandle(t, true)
+		f2 := createFunctionForSQLHandle(t, true)
+
+		functions, err := client.Functions.Show(ctx, sdk.NewShowFunctionRequest().WithLike(f1.Name))
+		require.NoError(t, err)
+
+		require.Equal(t, 1, len(functions))
+		require.Contains(t, functions, *f1)
+		require.NotContains(t, functions, *f2)
+	})
+
+	t.Run("show function for SQL: no matches", func(t *testing.T) {
+		functions, err := client.Functions.Show(ctx, sdk.NewShowFunctionRequest().WithLike(random.String()))
+		require.NoError(t, err)
+		require.Equal(t, 0, len(functions))
+	})
+
+	t.Run("describe function for SQL", func(t *testing.T) {
```

Comment:
is describe always empty ? or we can test it with some values and see that they're mapped correctly etc.

Reply:
added a few value tests
```
		require.Equal(t, "SQL", pairs["language"])
		require.Equal(t, "FLOAT", pairs["returns"])
		require.Equal(t, "3.141592654::FLOAT", pairs["body"])
		require.Equal(t, "(X FLOAT)", pairs["signature"])
```

Reply:
added a few value tests
		require.Equal(t, "SQL", pairs["language"])
		require.Equal(t, "FLOAT", pairs["returns"])
		require.Equal(t, "3.141592654::FLOAT", pairs["body"])
		require.Equal(t, "(X FLOAT)", pairs["signature"])